app/components/dex-create.jsx

```diff
@@ -32,8 +41,8 @@ export class DexCreate extends Component {
     const { createDex, redirect, session } = this.props;
     const title = this._title.value;
     const shiny = this._shiny.checked;
-    const generation = this._generation.value;
-    const region = parseInt(generation) === 7 ? 'alola' : 'national';
+    const generation = this.state.generation;
+    const region = this.state.region;
```

Comment:
instead of using `this.state` directly, we should destructure it just like what we do for `this.props` above. thats what we do [elsewhere](https://github.com/pokedextracker/pokedextracker.com/blob/master/app/components/account.jsx#L73-L74)

```js
const { createDex, redirect, session } = this.props;
const { generation, region } = this.state;
```